mayastor/src/bdev/nexus/nexus_share.rs

```diff
@@ -100,6 +100,15 @@ impl Nexus {
     pub async fn unshare(&mut self) -> Result<(), Error> {
         match self.nbd_disk.take() {
             Some(disk) => {
+                let sp = match self.share_protocol {
+                    ShareProtocol::Nbd => Some(ShareProtocol::Nvmf),
+                    ShareProtocol::Iscsi => Some(ShareProtocol::Nvmf),
+                    ShareProtocol::Nvmf => Some(ShareProtocol::Nvmf),
```

Comment:
Maybe we have a single function rather than duplicating this with the code around line 48? So that adding/removing a protocol can be done in fewer places.

Reply:
Actually, I suppose that's not worth changing ATM. My changes will affect that part of the code anyway.

Reply:
https://github.com/tjoshum/Mayastor/pull/3/commits/ca0905f02e30620700f17fcc6f942a4429211d4d